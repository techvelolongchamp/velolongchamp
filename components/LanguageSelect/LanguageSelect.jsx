import React from 'react'
import Link from 'next/link'
import { useRouter } from 'next/router'

import { LngContainer, SelectLng } from './LanguageSelect.styled'

const LanguageSelect = () => {
  const { locale, pathname, query } = useRouter()

  return (
    <LngContainer>
      {locale === 'fr' ? (
        <Link href={{ pathname, query }} locale="en">
          <SelectLng>EN</SelectLng>
        </Link>
      ) : (
        <Link href={{ pathname, query }} locale="fr">
          <SelectLng>FR</SelectLng>
        </Link>
      )}
    </LngContainer>
  )
}

export default LanguageSelect
