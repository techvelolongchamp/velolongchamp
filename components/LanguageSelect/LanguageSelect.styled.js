import styled from 'styled-components'

export const LngContainer = styled.div`
  align-items: center;
  display: flex;
  height: 100%;
  font-size: ${({ theme }) => theme.fonts.small};
  margin-left: ${({ theme }) => theme.spacing.xs};
`

export const SelectLng = styled.button`
  background: none;
  border: none;
  outline: none;
  padding: ${({ theme }) => theme.spacing.xs};
  width: 24px;

  &:hover {
    text-decoration: underline;
  }
`
