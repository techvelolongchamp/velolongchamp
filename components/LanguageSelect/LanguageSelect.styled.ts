import styled from 'styled-components'

export const LngContainer = styled.div`
  align-items: center;
  display: flex;
  height: 100%;
  font-size: ${({ theme }) => theme.fonts.small};
  margin: 0 ${({ theme }) => theme.spacing.xs};
`

export const SelectLng = styled.button`
  background: none;
  border: none;
  outline: none;
  color: black;
  padding: ${({ theme }) => theme.spacing.xs};
  text-decoration: none;

  &:hover {
    text-decoration: underline;
  }
`
