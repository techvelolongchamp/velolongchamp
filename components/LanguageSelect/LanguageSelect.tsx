import React from 'react'
import Link from 'next/link'
import { useRouter } from 'next/router'

import { LngContainer, SelectLng } from './LanguageSelect.styled'

const LanguageSelect = () => {
  const { locale, pathname, query } = useRouter()

  return (
    <LngContainer>
      {locale === 'fr' ? (
        <Link href={{ pathname, query }} locale="en" passHref>
          <SelectLng id="select-en" data-testid="select-en">
            EN
          </SelectLng>
        </Link>
      ) : (
        <Link href={{ pathname, query }} locale="fr" passHref>
          <SelectLng id="select-fr" data-testid="select-fr">
            FR
          </SelectLng>
        </Link>
      )}
    </LngContainer>
  )
}

export default LanguageSelect
