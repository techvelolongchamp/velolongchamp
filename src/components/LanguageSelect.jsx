import React, { useState } from 'react'
import styled from 'styled-components'
import { useTranslation } from 'react-i18next'

const LngContainer = styled.div`
  align-items: center;
  display: flex;
  height: 100%;
  margin-left: ${({ theme }) => theme.spacing.xs};
`

const SelectLng = styled.button`
  background: none;
  border: none;
  outline: none;
  padding: ${({ theme }) => theme.spacing.xs};
  width: 24px;

  &:hover {
    text-decoration: underline;
  }
`

const LanguageSelect = () => {
  const { i18n } = useTranslation()
  const [language, setLanguage] = useState(i18n.language || 'fr-FR')

  const handleSelect = function() {
    setLanguage(this.value)
    i18n.changeLanguage(this.value)
  }

  return (
    <LngContainer>
      {language === 'fr-FR' || language === 'fr' ? (
        <SelectLng onClick={handleSelect.bind({ value: 'en' })}>EN</SelectLng>
      ) : (
        <SelectLng onClick={handleSelect.bind({ value: 'fr' })}>FR</SelectLng>
      )}
    </LngContainer>
  )
}

export default LanguageSelect
