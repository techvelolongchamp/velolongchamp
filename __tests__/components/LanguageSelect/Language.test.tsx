import React from 'react'

import { RenderWithProvider as render } from '../../provider'

import LanguageSelect from '../../../components/LanguageSelect/LanguageSelect'

const useRouter = jest.spyOn(require('next/router'), 'useRouter')

describe('LanguageSelect unit tests', () => {
  it('should render correctly', () => {
    useRouter.mockImplementation(() => ({
      locale: 'fr',
      pathname: '/',
    }))
    const { container, getByTestId } = render(<LanguageSelect />)
    const select = getByTestId('select-en')
    expect(container).toMatchSnapshot()

    expect(select).toHaveAttribute('href', '/')
  })

  it('should render correctly - en', () => {
    const push = jest.fn()
    useRouter.mockImplementation(() => ({ locale: 'en', pathname: '/', push }))
    const { container, getByTestId } = render(<LanguageSelect />)

    const select = getByTestId('select-fr')
    expect(container).toMatchSnapshot()

    expect(select).toHaveAttribute('href', '/')
  })
})
