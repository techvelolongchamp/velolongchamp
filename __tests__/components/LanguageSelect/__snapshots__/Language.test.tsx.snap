// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelect unit tests should render correctly - en 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  font-size: 12px;
  margin-left: 5px;
}

.c1 {
  background: none;
  border: none;
  outline: none;
  padding: 5px;
  width: 24px;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1"
      data-testid="select-fr"
      href="/"
      id="select-fr"
    >
      FR
    </button>
  </div>
</div>
`;

exports[`LanguageSelect unit tests should render correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  font-size: 12px;
  margin-left: 5px;
}

.c1 {
  background: none;
  border: none;
  outline: none;
  padding: 5px;
  width: 24px;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div>
  <div
    class="c0"
  >
    <button
      class="c1"
      data-testid="select-en"
      href="/"
      id="select-en"
    >
      EN
    </button>
  </div>
</div>
`;
